index.js

```diff
@@ -1,24 +1,10 @@
-const express = require('express');
-
-const shopifyAuthRouter = require('./routes/shopifyAuth');
-const shopifyApiProxy = require('./routes/shopifyApiProxy');
-
-const withShop = require('./middleware/withShop');
-const withWebhook = require('./middleware/webhooks');
-
-const shopStore = require('./shopStore');
-
-module.exports = {
-  withShop: withShop,
-  withWebhook: withWebhook,
-  shopStore: shopStore
-};
-
-module.exports.shopifyRouter = function(shopifyConfig) {
-  const router = express.Router();
-
-  router.use('/auth/shopify', shopifyAuthRouter(shopifyConfig));
-  router.use('/api', withShop({ redirect: false }), shopifyApiProxy);
-
-  return router
+const createRouter = require('./routes');
+const createMiddleware = require('./middleware');
+const {MemoryStrategy} = require('./strategies');
+
+module.exports = function shopify(shopifyConfig = {shopStore: new MemoryStrategy()}) {
```

Comment:
Changed because
- We need to pass the config down to each child module
- Consistency is good, putting router creation into an index file is a slight bump to consistency
- Two top level keys makes things easy to grok

Reply:
Love the refactor but wouldn't this make `shopStore` set to `undefined` if I passed in a config object without that prop?